SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp

```diff
@@ -76,10 +76,9 @@ EPushRPCResult SpatialRPCService::PushRPC(const Worker_EntityId EntityId, const
 
 	if (RPCRingBufferUtils::ShouldQueueOverflowed(Type) && ClientServerRPCs.ContainsOverflowedRPC(EntityType))
 	{
-		if (EventTracer != nullptr)
+		if (EventTracer != nullptr && EventTracer->IsEnabled() && PendingPayload.SpanId.IsSet())
```

Comment:
@MatthewSandfordImprobable I should just trace here right?

Reply:
Yeah but I can patch this in my PR if you like. I will probs be removing the optionals so...

Reply:
I'll revert this - some ideas here conflict with your PR